service/controller/v1/resource/configmap/configmap.go

```diff
@@ -76,3 +81,63 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
+	currentCM, err := r.getCurrentState(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	desiredCM, err := r.getDesiredState(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cm := newConfigMapToUpdate(currentCM, desiredCM)
+
+	{
+		// currentCM is used in the log messages because cm can be nil.
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating ConfigMap %#q in namespace %#q", currentCM.GetName(), currentCM.GetNamespace()))
+
+		if cm == nil {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ConfigMap %#q in namespace %#q is up to date", currentCM.GetName(), currentCM.GetNamespace()))
```

Comment:
This does not seem accurate.
`newConfigMapToUpdate` checks if current and desired name / namespace are equal and returns `nil` if they aren't.
This does not imply in any way that the CM is up to date.

Reply:
This is because `newConfigMapToUpdate` is generic. If `getCurrentState` and `getDesiredState` don't return CM with the same name and namespace then we have a bug in the code.

I understand where you come from but I wouldn't bother unless it's really important to you or you have concrete suggestions how to solve that.

Reply:
Why then even check for name / namespace? 

Because like this, even if we do have a bug, we will not suspect it there because of the log message. So don't check name/namespace and have a failure that is visible.

Reply:
Because I copied it from here https://github.com/giantswarm/operatorkit/blob/master/resource/k8s/configmapresource/new_config_map_to_update.go

Reply:
I couldn't use the resource from the operatorkit because the deletion here is handled differently.

Reply:
Your words are a mystery to me.

It's fine that you copied it, but why not make the world a better place here and not check for name & namespace in that function? Maybe I am just not getting it :D

Reply:
We talked I removed name & namespace checks from newConfigMapToUpdate. In case of a bug we should have easier time noticing it. Thanks Marcel.